### ISSUE TITLE
pdnsutil edit-zone does not handle EDITOR='sed -i "command"' properly

### ISSUE THREAD

**Issue by pieterlexis** [PowerDNS/pdns#3829]
from IRC:
```
09:21:04 < lieter> also... EDITOR='sed "s/foo/bar"' pdnsutil edit-zone
09:21:14 < Habbie> oh snap
09:21:27 < lieter> no clue if it works with pdnsutil
09:21:38 < lieter> but it worked for some other things I tried
09:21:53 < Habbie> 'No change to file'
09:21:57 < Habbie> it confuses it in some way
09:21:58 < vhasi> -i
09:22:11 < Habbie> better but still wrong
09:22:14 < Habbie> maybe we do something wrong
09:22:23 < lieter> this is fully possible
09:22:46 < Habbie> the first version did show me the 'changed' zone file
09:22:48 < Habbie> on stdout
09:22:52 < Habbie> never went to pdnsutil i think
```

**Comment by pieterlexis** [comment]
e.g. the `virsh edit <domain>` command works properly.

[Source of virsh-edit](http://libvirt.org/git/?p=libvirt.git;a=blob;f=tools/virsh-edit.c;h=1b39cb7d4525e2ade5e2c94d6134a134c7b72e69;hb=HEAD)

[Implementation of the edit](http://libvirt.org/git/?p=libvirt.git;a=blob;f=tools/vsh.c;h=605c5745c4a2778bd0c2a5e73817169c5ad759d9;hb=HEAD#l2342)

**Comment by hnsk** [comment]
I just spent quite a while investigating this. If you do the following it works
```
EDITOR='sleep 1; sed -i "s/foo/bar/"' pdnsutil edit-zone example.com
```

The problem is that pdnsutil compares ctime of the file before and after the change and it uses 1 second precision (also you can't have more accurate on ext3 for example).

We discussed the possibility of just removing this check and always doing the diff as it's pretty lightweight since the zone is loaded anyway.

**Comment by rgacogne** [comment]
Fixed by #4183 (thanks!).
